sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable, Inject } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Institution, User } from "../../database/entities";
+import { Connection } from "typeorm";
+import { UserInfo } from "../../types";
+import { CreateInstitutionDto } from "../../route-controllers/institution/models/institution.dto";
+import { LoggerService } from "../../logger/logger.service";
+import { InjectLogger } from "../../common";
+
+@Injectable()
+export class InstitutionService extends RecordDataModelService<Institution> {
+  @InjectLogger()
+  logger: LoggerService;
+  constructor(@Inject("Connection") connection: Connection) {
+    super(connection.getRepository(Institution));
+    this.logger.log("[Created]");
+  }
+
+  async getInstitutionByUserName(userName: string): Promise<Institution> {
```

Comment:
I don't believe that we are going to have a one-to-one relationship between institutions and users.
It will probably require further discussion.

Reply:
DOne